apps/domain/lib/repos/repo.ex

```diff
@@ -9,26 +9,35 @@ defmodule Domain.Repos.Repo do
 
   @derive {Poison.Encoder, except: [:tags]}
 
-  @enforce_keys [:name, :owner]
+  # 1 Hour in ms
+  @one_hour 3_600_000
+
+  @enforce_keys [:id, :url, :polling_interval]
+  @type interval :: non_neg_integer()
+  @type id :: non_neg_integer()
   @type t :: %__MODULE__{
+          id: id(),
           name: String.t(),
           owner: String.t(),
+          url: String.t(),
+          # In milliseconds
+          polling_interval: interval(),
           tags: list(Tag.t()),
           tasks: list(Task.t())
         }
 
-  defstruct name: nil, owner: nil, tags: [], tasks: []
+  defstruct id: nil, name: nil, owner: nil, url: nil, polling_interval: nil, tags: [], tasks: []
 
-  @spec new(binary()) :: Repo.t()
-  def new(url) do
+  @spec new(id(), String.t(), interval()) :: Repo.t()
+  def new(id, url, interval \\ @one_hour) do
     %{path: path} = URI.parse(url)
```

Comment:
I know it's the default in Ecto - but do we need autogenerated id()? A Repo can be uniquely identified by the { owner (user who added it to the system), URL}

Reply:
I thought so, but as we are fetching tasks and polling/storing tags i thought it would be easier to work with an id, but yeah, there is a unique index on repository_url. do you prefer to get rid of that id and use `repository_url` to make the association? (all the queries are going to be exec via RPC in the Admin App) so it would be something like `all_tasks_by_repository_url` or similar

Reply:
Perhaps make the unique index on URL/user ?

Reply:
I don't think we need a unique index on URL/user because there shouldn't be 2 repositories with the same URL, there should be a repository an many tasks, so if that repository already exists user would add a new task instead of creating a new repository. or what do you think?

Reply:
Yes right now the user who adds the repo is the owner of the repo, but everyone can add tasks to it